### PR TITLE
Fix outdated doc ref in comment

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Fix outdated doc references in comment.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/core/v1/validation/validation.go [hunk 1]
```diff
@@ -107,7 +107,8 @@ func ValidateNonnegativeQuantity(value resource.Quantity, fldPath *field.Path) f
 }
 
 // Validate compute resource typename.
-// Refer to docs/design/resources.md for more details.
+// Refer to https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/resources.md
```
Comment (deads2k) [inline pkg/apis/core/v1/validation/validation.go:110]: this file doesn't appear to match content about a quota resource name

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -4958,7 +4960,8 @@ func validateContainerResourceName(value string, fldPath *field.Path) field.Erro
 }
 
 // Validate resource names that can go in a resource quota
-// Refer to docs/design/resources.md for more details.
+// Refer to https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/resources.md
```
Comment (deads2k) [inline pkg/apis/core/validation/validation.go:4963]: this looks like the wrong file ref for all of the methods in this file.

File: staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go [hunk 3]
```diff
@@ -32,7 +32,8 @@ import (
 // certain fields with metadata that indicates whether the elements of JSON
 // lists should be merged or replaced.
 //
-// For more information, see the PATCH section of docs/devel/api-conventions.md.
+// For more information, see the PATCH section of
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go:35]: please use a link that pins to a sha.

**Comment by deads2k** [comment]
/assign

get me on slack once it's been updated.

File: pkg/apis/core/v1/validation/validation.go (hunk 1 above)
Comment (lingsamuel) [reply pkg/apis/core/v1/validation/validation.go:110]: Nope, they are (almost) same file.
https://github.com/kubernetes/kubernetes/blob/bff3cb4df68d699285285c09c1c46be69ef1465a/docs/design/resources.md
vs
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/resources.md

Resource type name [def here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/resources.md#resource-types).

File: staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go (hunk 3 above)
Comment (lingsamuel) [reply staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go:35]: done

**Comment by deads2k** [comment]
so it is

/lgtm
/approve
/ok-to-test

File: pkg/apis/core/v1/validation/validation.go [hunk 4]
```diff
@@ -107,7 +107,8 @@ func ValidateNonnegativeQuantity(value resource.Quantity, fldPath *field.Path) f
 }
 
 // Validate compute resource typename.
-// Refer to docs/design/resources.md for more details.
+// Refer to https://github.com/kubernetes/community/blob/01036e0d3f4230b2ba8affbb151ddd58d4706aa3/contributors/design-proposals/scheduling/resources.md
```
Comment (liggitt) [inline pkg/apis/core/v1/validation/validation.go:110]: should this refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ instead?

Comment (lingsamuel) [reply pkg/apis/core/v1/validation/validation.go:110]: I am afraid this page doesn't contain exact rule like [this section](https://github.com/kubernetes/community/blob/01036e0d3f4230b2ba8affbb151ddd58d4706aa3/contributors/design-proposals/scheduling/resources.md#resource-types)

**Comment by dims** [comment]
@lingsamuel will you be revisiting this PR? if there is stuff missing in our public documentation, we should update that (and keep @liggitt 's suggestion)

**Pull request closed without merging**